tests/unit/test_s3_parquet.py

```diff
@@ -886,3 +886,73 @@ def test_chunked_columns(path, columns, chunked):
         df2 = pd.concat(list(df2), ignore_index=True)
 
     assert df[columns].shape if columns else df.shape == df2.shape
+
+
+@pytest.mark.xfail(
+    is_ray_modin,
+    raises=TypeError,
+    reason="Ray Modin cannot serialize Pyarrow crytography objects since they are C++ objects",
+)
+@pytest.mark.parametrize("max_rows_by_file", [1, 2, 100])
+@pytest.mark.parametrize("chunked", [False, True, 2])
+@pytest.mark.parametrize("columns", [["c0", "c1", "c3"], ["c0"], ["c0", "c1"]])
+def test_write_to_parquet_with_client_encryption_config(
+    path, bucket, kms_key_id, max_rows_by_file, chunked, columns, client_encryption_materials
+):
+    df = pd.DataFrame({"c0": [0, 1, 2, 3, 4], "c1": [2, 3, 4, 5, 6], "c3": [3, 4, 5, 6, 7]}, dtype="Int64")
+    path_file = f"{path}0.parquet"
+    crypto_factory, kms_connection_config, encryption_config = client_encryption_materials
+    wr.s3.to_parquet(
+        df,
+        path_file,
+        pyarrow_additional_kwargs={
+            "crypto_factory": crypto_factory,
+            "kms_connection_config": kms_connection_config,
+            "encryption_config": encryption_config,
+        },
+        max_rows_by_file=max_rows_by_file,
+    )
+    df_out = wr.s3.read_parquet(
+        path,
+        decryption_configurations={"crypto_factory": crypto_factory, "kms_connection_config": kms_connection_config},
+        chunked=chunked,
+    )
+    if chunked:
+        df_out = pd.concat(list(df_out), ignore_index=True)
+    assert_pandas_equals(df, df_out)
+
+
+@pytest.mark.xfail(
+    is_ray_modin,
+    raises=TypeError,
+    reason="Ray Modin cannot serialize Pyarrow crytography objects since they are C++ objects",
+)
+@pytest.mark.xfail(
+    not is_ray_modin,
+    raises=OSError,
+    reason="Issue with pyarrow read_table when using client side encryption, see https://github.com/apache/arrow/issues/39645",
+)
+@pytest.mark.parametrize("columns", [["c0", "c1"], ["c0"]])
+def test_read_parquet_table_with_client_side_encryption(
+    path, glue_database, glue_table, kms_key_id, columns, client_encryption_materials
+):
+    df = pd.DataFrame({"c0": [0, 1, 2], "c1": [0, 1, 2], "c2": [0, 0, 1]})
+    crypto_factory, kms_connection_config, encryption_config = client_encryption_materials
+    wr.s3.to_parquet(
+        df,
+        path,
+        dataset=True,
+        database=glue_database,
+        table=glue_table,
+        pyarrow_additional_kwargs={
+            "crypto_factory": crypto_factory,
+            "kms_connection_config": kms_connection_config,
+            "encryption_config": encryption_config,
+        },
+    )
+    df_out = wr.s3.read_parquet_table(
+        table=glue_table,
+        database=glue_database,
+        decryption_configurations={"crypto_factory": crypto_factory, "kms_connection_config": kms_connection_config},
+    )
+    assert df_out.shape == (3, 3)
```

Comment:
It seems the `OSError` issue only happens when we pass `validate_schema=True`. This parameter is True by default when using `read_parquet_table`, but False by default when we invoke `read_parquet`. It would be good to make the condition of the `OSError` even more specific.

For example:

```python
@pytest.mark.xfail(
    is_ray_modin,
    raises=TypeError,
    reason="Ray Modin cannot serialize Pyarrow crytography objects since they are C++ objects",
)
@pytest.mark.parametrize("columns", [["c0"], ["c0", "c1"]])
@pytest.mark.parametrize(
    "validate_schema",
    [
        pytest.param(False),
        pytest.param(
            True,
            marks=pytest.mark.xfail(
                not is_ray_modin,
                raises=OSError,
                reason="Issue with pyarrow read_table when using client side encryption, see https://github.com/apache/arrow/issues/39645",
            ),
        ),
    ],
)
def test_read_parquet_table_with_client_side_encryption(
    path, glue_database, glue_table, kms_key_id, columns, validate_schema, client_encryption_materials
):
    df = pd.DataFrame({"c0": [0, 1, 2], "c1": [0, 1, 2], "c2": [0, 0, 1]}, dtype="Int64")
    crypto_factory, kms_connection_config, encryption_config = client_encryption_materials
    wr.s3.to_parquet(
        df,
        path,
        dataset=True,
        database=glue_database,
        table=glue_table,
        pyarrow_additional_kwargs={
            "crypto_factory": crypto_factory,
            "kms_connection_config": kms_connection_config,
            "encryption_config": encryption_config,
        },
    )
    df_out = wr.s3.read_parquet_table(
        table=glue_table,
        database=glue_database,
        decryption_configurations={"crypto_factory": crypto_factory, "kms_connection_config": kms_connection_config},
        validate_schema=validate_schema,
        columns=columns,
    )
    assert_pandas_equals(df[columns], df_out)
```